frontend/catalyst/qjit_device.py

```diff
@@ -218,3 +220,96 @@ def _decomp_to_unitary(self, *_args, **_kwargs):
 
         self.check_validity(expanded_tape.operations, [])
         return expanded_tape
+
+
+class QJITDeviceNewAPI(qml.devices.Device):
```

Comment:
Let me double check: do we really need a separate class here? Why can't we go with e.g.
``` python
class A: pass
class B: pass

def qjit_device_factory(base, *args, **kwargs):
  class QJITDevice(base):
    # The implementation hides the differences between old and new PL devices, but provides a
    # common API for Catalyst needs
    def __init__(self, a, b, c):
       if isinstance(base, A):
          print("Init A", base, a, b, c)
       elif isinstance(base, B):
          print("Init B", base, a, b, c)

  return QJITDevice(*args, **kwargs)

qjit_device1 = qjit_device_factory(A, 1, 2, 3)
qjit_device2 = qjit_device_factory(B, 1, 2, 3)
```
```

Reply:
The two base classes are very different, so I prefer to have an explicit difference even in the QJIT device. Different methods are to be implemented and therefore to be used in the frontend (expand vs preprocess for example). I prefer not to hide those differences.